runtime/common/src/claims.rs

```diff
@@ -405,7 +416,13 @@ decl_module! {
 		) {
 			T::MoveClaimOrigin::try_origin(origin).map(|_| ()).or_else(ensure_root)?;
 
-			Claims::<T>::take(&old).map(|c| Claims::<T>::insert(&new, c));
+			Claims::<T>::take(&old).map(|c| Claims::<T>::mutate(&new, |maybe_overwrite| {
+				if let Some(overwrite) = maybe_overwrite {
+					Total::<T>::mutate(|total| *total = *total - *overwrite + c);
+				}
+
+				*maybe_overwrite = Some(c);
+			}));
 			Vesting::<T>::take(&old).map(|c| Vesting::<T>::insert(&new, c));
 			Signing::take(&old).map(|c| Signing::insert(&new, c));
 			maybe_preclaim.map(|preclaim| Preclaims::<T>::mutate(&preclaim, |maybe_o|
```

Comment:
what if the new ethereum account is also preclaim ? (i.e. it exist an account where Preclaim(account) == new)